### PR TITLE
mobile,api: new deposit tab design

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
Screenshot 2024-03-31 at 11 25 52 PM
Screenshot 2024-03-31 at 11 25 45 PM
Screenshot 2024-03-31 at 11 25 41 PM
Screenshot 2024-03-31 at 11 25 37 PM

### CODE REVIEW

File: apps/daimo-mobile/src/view/screen/DepositScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+import { useWithAccount } from "../shared/withAccount";
+
+export default function DepositScreen() {
+  const Inner = useWithAccount(DepositScreenInner);
+  return <Inner />;
+}
+
+function DepositScreenInner({ account }: { account: Account }) {
+  return (
+    <View style={{ flex: 1 }}>
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/DepositScreen.tsx:33]: `style={ss.container.screen}`?

File: apps/daimo-mobile/src/view/screen/DepositScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        <ScreenHeader title="Deposit or Withdraw" onExit={useExitToHome} />
+      </View>
+      <ScrollView>
+        <CoverGraphic type="deposit" />
+        <Spacer h={8} />
+        <DepositList account={account} />
+        <Spacer h={16} />
+        <WithdrawList />
+      </ScrollView>
+    </View>
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/DepositScreen.tsx:44]: super clean 👍

File: apps/daimo-mobile/src/view/screen/DepositScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        logo: rec.logo || defaultLogo,
+        isExternal: true,
+        onClick: () => openExchange(rec.url),
+      }))
+    );
+  }
+
+  return (
+    <View style={{ marginHorizontal: 16 }}>
+      <DaimoText style={{ ...ss.text.body, color: color.gray3 }}>
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/DepositScreen.tsx:89]: `<TextBody color={color.gray3}>`

File: apps/daimo-mobile/src/view/screen/DepositScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+  const width = useWindowDimensions().width;
+
+  return (
+    <TouchableHighlight
+      onPress={onClick}
+      {...touchHighlightUnderlay.subtle}
+      style={{ marginHorizontal: -16 }}
+    >
+      <View style={styles.optionRow}>
+        <View style={{ ...styles.optionRowLeft, maxWidth: width - 160 }}>
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/DepositScreen.tsx:153]: you may be able to make `optionRowRight` set `width: 160` or similar and avoid using window dimensions

File: apps/daimo-mobile/src/view/shared/CoverGraphic.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { View, StyleSheet, Image } from "react-native";
+
+import DepositCover from "../../../assets/deposit-cover.png";
+import InviteCover from "../../../assets/invite-cover.png";
+
+export function CoverGraphic({ type }: { type: "invite" | "deposit" }) {
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/shared/CoverGraphic.tsx:6]: very clean

File: apps/daimo-mobile/src/view/sheet/DepositAddressBottomSheet.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+import { env } from "../../logic/env";
+import { Account } from "../../model/account";
+import { CheckLabel } from "../shared/Check";
+import { ScreenHeader } from "../shared/ScreenHeader";
+import Spacer from "../shared/Spacer";
+import { color, ss, touchHighlightUnderlay } from "../shared/style";
+import { DaimoText, TextBold, TextLight, TextPara } from "../shared/text";
+import { useWithAccount } from "../shared/withAccount";
+
+export function DepositAddressBottomSheet() {
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/sheet/DepositAddressBottomSheet.tsx:18]: recommend a 1-sentence comment above components like this saying what it does

eg > `// Explains how to deposit money directly to your Daimo address`

File: apps/daimo-mobile/src/view/sheet/DepositAddressBottomSheet.tsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+  const dispatcher = useContext(DispatcherContext);
+
+  const { tokenSymbol, chainL2 } = env(
+    daimoChainFromId(account.homeChainId)
+  ).chainConfig;
+
+  const [check1, setCheck1] = useState(false);
+  const [check2, setCheck2] = useState(false);
+
+  assert(tokenSymbol === "USDC", "Unsupported coin: " + tokenSymbol);
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/sheet/DepositAddressBottomSheet.tsx:33]: good defensive coding

File: apps/daimo-mobile/src/view/sheet/FarcasterBottomSheet.tsx [hunk 8]
```diff
@@ -105,6 +105,7 @@ function FarcasterQRButton({ url }: { url: string }) {
         title="OPEN IN WARPCAST"
         onPress={openInWarpcast}
       />
+      <Spacer h={16} />
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/sheet/FarcasterBottomSheet.tsx:108]: might not be necessary anymore. FarcasterBottomSheet is now fixed height

File: apps/daimo-mobile/src/view/sheet/GlobalBottomSheet.tsx [hunk 9]
```diff
@@ -42,6 +44,12 @@ const bottomSheetSettings = {
   ownRequest: {
     enableSwipeClose: true,
   },
+  withdrawInstructions: {
+    enableSwipeClose: true,
+  },
+  depositAddress: {
+    enableSwipeClose: true,
+  },
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/sheet/GlobalBottomSheet.tsx:52]: 🔥

**Review by dcposch — APPROVED** [review]
🔥

File: apps/daimo-mobile/src/view/screen/DepositScreen.tsx (hunk 1 above)
Comment (nalinbhardwaj) [reply apps/daimo-mobile/src/view/screen/DepositScreen.tsx:33]: not the same effect, don't want padding

File: apps/daimo-mobile/src/view/screen/DepositScreen.tsx (hunk 4 above)
Comment (nalinbhardwaj) [reply apps/daimo-mobile/src/view/screen/DepositScreen.tsx:153]: tried some variants but they all make it bleed offscreen, or it ends up breaking the underlay, maybe I'm missing something obvious

Simulator Screenshot - iPhone 14 Pro - 2024-04-04 at 12 13 03

File: apps/daimo-mobile/src/view/sheet/FarcasterBottomSheet.tsx (hunk 8 above)
Comment (nalinbhardwaj) [reply apps/daimo-mobile/src/view/sheet/FarcasterBottomSheet.tsx:108]: dropped

File: apps/daimo-mobile/src/view/sheet/DepositAddressBottomSheet.tsx (hunk 6 above)
Comment (nalinbhardwaj) [reply apps/daimo-mobile/src/view/sheet/DepositAddressBottomSheet.tsx:18]: done

**Pull request merged**